src/fmu/dataio/providers/_objectdata.py

```diff
@@ -0,0 +1,676 @@
+            logger.debug("arrow")
+            from pyarrow import Table
+
+            assert isinstance(self.obj, Table)
+            columns = self.obj.column_names
+        logger.debug("Available columns in table %s ", columns)
+        return columns
+
+    def _derive_index(self) -> list[str]:
+        """Derive table index"""
+        # This could in the future also return context
+        columns = self._get_columns()
+        index = []
+
+        if self.dataio.table_index is None:
+            logger.debug("Finding index to include")
+            for context, standard_cols in STANDARD_TABLE_INDEX_COLUMNS.items():
+                for valid_col in standard_cols:
+                    if valid_col in columns:
+                        index.append(valid_col)
+                if index:
+                    logger.info("Context is %s ", context)
+            logger.debug("Proudly presenting the index: %s", index)
+        else:
+            index = self.dataio.table_index
+
+        if "REAL" in columns:
+            index.append("REAL")
+        self._check_index(index)
+        return index
+
+    def _check_index(self, index: list[str]) -> None:
+        """Check the table index.
+        Args:
+            index (list): list of column names
+
+        Raises:
+            KeyError: if index contains names that are not in self
+        """
+
+        not_founds = (item for item in index if item not in self._get_columns())
+        for not_found in not_founds:
+            raise KeyError(f"{not_found} is not in table")
+
+    def _derive_spec_and_bbox(self) -> SpecificationAndBoundingBox:
+        """Process/collect the data items for Arrow table."""
+        logger.info("Process data metadata for arrow (tables)")
+        from pyarrow import Table
+
+        assert isinstance(self.obj, Table)
+        return SpecificationAndBoundingBox(
+            spec=specification.TableSpecification(
+                columns=list(self.obj.column_names),
+                size=self.obj.num_columns * self.obj.num_rows,
+            ).model_dump(
+                mode="json",
+                exclude_none=True,
+            ),
+            bbox={},
+        )
+
+    def _derive_objectdata(self) -> DerivedObjectDescriptor:
+        spec, bbox = self._derive_spec_and_bbox()
+        return DerivedObjectDescriptor(
+            table_index=self._derive_index(),
+            subtype="ArrowTable",
+            classname="table",
+            layout="table",
+            efolder="tables",
+            fmt=(fmt := self.dataio.arrow_fformat),
+            extension=self._validate_get_ext(fmt, "ArrowTable", ValidFormats().table),
+            spec=spec,
+            bbox=bbox,
+        )
+
+
+@dataclass
+class ExistingDataProvider(ObjectDataProvider):
+    """These functions should never be called because derive_metadata will populate the
+    object data from existing metadata, by calling _derive_from_existing, and return
+    before calling them."""
```

Comment:
This works for now but I will change it later. If complete metadata already exists it feels a bit unnecessary to step into a class that is meant to derive it; it should just populate the class (or, eventually, just instantiate a Pydantic model with it)